### ISSUE TITLE
Question about Spatial Correlation Implementation

### ISSUE THREAD

**Issue by hshustc** [anguyen8/deepface-emd#5]
In the implementation for Spatial Correlation (emd_similarity function in emd.py), u and v are computed based on the distance instead of the similarity. Is it correct?
```
        elif method == 'sc':
            u = torch.sum(dis, 2)
            u = u / (u.sum(dim=1, keepdims=True) + 1e-7)
            v = torch.sum(dis, 1)
            v = v / (v.sum(dim=1, keepdims=True) + 1e-7)
```

**Comment by haithanhp** [comment]
Hi @hshustc,

Thanks for your question. u and v can be computed from the similarity but based on our observations, the accuracy is bad with the current implementation. That is the reason why we used distances. It may be because of relative quantities in Sinkhorn optimization.

**Comment by hshustc** [comment]
That is strange. As far as I can see, u and v denote the weights (or importance)  of each patch. For other implementations, the patches with higher similarity are assigned larger weights, but it is the other way around for Spatial Correlation referring to the current implementation. Could you please give more explanation? Thanks a lot.

**Comment by anguyen8** [comment]
@hshustc Thank you for the great catch!

1. Yes, we will correct Eq. 8 in the paper to reflect that the Spatial Correlation version that we used actually uses `dis` (cosine distance) instead of `sim` (cosine similarity) in the [code](https://github.com/anguyen8/deepface-emd/blob/main/utils/emd.py#L34-L37).

2. @haithanhp was experimenting with both (`dis` and `sim`) but did not have the time to investigate why `dis` yields better face identification results when running the DeepFace-EMD. I don't want to make a claim without evidence here so I'd suggest, if you are interested in using SC, you should try both versions. :D 

cheers!
